Source/WTF/wtf/SystemTracing.h

```diff
@@ -328,6 +347,31 @@ enum WTFOSSignpostType {
 #define WTFEmitSignpostIndirectlyWithType(type, pointer, name, timeDelta, format, ...) \
     os_log(WTFSignpostLogHandle(), "type=%d name=%d p=%" PRIuPTR " ts=%llu " format, type, WTFOSSignpostName ## name, reinterpret_cast<uintptr_t>(pointer), WTFCurrentContinuousTime(timeDelta), ##__VA_ARGS__)
 
+#elif USE(SYSPROF_CAPTURE)
+
+#define WTFEmitSignpost(pointer, name, ...) \
+    WTF::SysprofAnnotator::instantMark(pointer, _STRINGIFY(name), " " __VA_ARGS__)
```

Comment:
Instead of using static members that get the instance, they could be instance memebers and the callers would do `WTF::SysprofAnnotator::singleton().foo()`, omitting the pointer param if we don't use it.